Docs/simplifying docs (#927)
terramate-io/terramate: docs/getting-started/index.md

This tutorial aims to introduce you to the basic concepts of [Terramate](https://github.com/mineiros-io/terramate), and demonstrate how a filesystem-oriented code generator can improve the management of your Terraform code efficiently.

To follow this tutorial, you will need local installations of [Terramate](../installation.md), [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli), and [Git](https://git-scm.com/downloads). 

Note that you won't need a cloud account, as we'll be using the `local_file` resource to create a static site that demonstrates Terramate's fundamental principles.

> We launched a [Terramate Discord Server](https://terramate.io/discord) in case you have any additional questions or issues running the examples in this guide.
>
> 👉 [https://terramate.io/discord](https://terramate.io/discord)

## Set up Terramate

> If you haven’t installed Terramate yet, please refer to the [installation](../installation.md) section for various  options. 

> To begin, create a new directory, and within it, create a file named `terramate.tm.hcl` with the following content:

Reviewer: ```suggestion
To begin, create a new directory that contains a file named `terramate.tm.hcl` with the following content:
```
Just an idea.. not mandatory to apply this. Reads easier from my perspective

}
```

Terramate recognizes files with the `.tm` or `.tm.hcl` extension, and any file containing a `stack {}` block is considered a stack. A stack is simply a directory where Terramate generates Terraform files. 

> You can create the `stack.tm.hcl` file manually without an ID or by running the `terramate create` command, but generating unique IDs for stacks is recommended. Unique IDs enable direct stack referencing instead of using relative paths, which can make refactoring difficult.

Reviewer: ```suggestion
> **Note:** You can manually create the `stack.tm.hcl` file without an ID or by running the `terramate create` command.
> However, generating unique IDs for stacks is recommended since they enable direct stack referencing instead of relative paths, making refactoring difficult.
```

I would add this as a quote or so and make sure that the user understands it's optional but not the desired way

Author: Yes, this makes sense. Get readers' attention and conveys the message easily.

}
```

> The name `globals.tm.hcl` is not special, and the content could be included in an existing `terramate.tm.hcl` file. All Terramate files are merged during runtime, similar to how Terraform merges `.tf` files. This simplicity provides great flexibility.

Reviewer: ```suggestion
The name `globals.tm.hcl` is not required, and the content could be included in an existing `terramate.tm.hcl` file. All Terramate files are merged during runtime, similar to how Terraform merges `.tf` files. This simplicity provides great flexibility.
```

Maybe required makes more sense than special? Just an idea

```bash
$ terramate run terraform init
2023-05-11T12:58:43+01:00 ERR outdated code found action=checkOutdatedGeneratedCode() filename=dev/mysite/mysite.tf
2023-05-11T12:58:43+01:00 ERR outdated code found action=checkOutdatedGeneratedCode() filename=prod/mysite/mysite.tf
2023-05-11T12:58:43+01:00 FTL please run: 'terramate generate' to update generated code error="outdated generated code detected" action=checkOutdatedGeneratedCode()
```

Whoops, we have errors!

We should have run `terramate generate` before `terramate run`. Thankfully, Terramate detected that the generated code needed to be updated and prevented us from running Terraform with outdated code.

To fix this, run `terramate generate`, followed by `terramate run terraform init` again. Terraform should now initialize sequentially in dev and prod. By default, terramate run executes in filesystem order, but [Terramate can control the order of execution](../orchestration/index.md#stacks-ordering) if needed.

Now run:

```bash
$ terramate run terraform apply
```

After approval, the rendered HTML files should be located at `/tmp/tfmysite/<env>/mysite/index.html`

### Terramate Orchestration with Change Detection

> To conclude this introduction, let's explore globals precedence and one of Terramate's most powerful features: change detection. Change detection is designed for CI-CD pipelines and requires a working Git repository with a remote. In your project's root, run:

Reviewer: ```suggestion
To conclude this introduction, let's explore globals precedence and one of Terramate's most powerful features: change detection. [Change detection](../change-detection/index.md) is designed for CI-CD pipelines and requires a working Git repository with a remote. In your project's root, run:
```
